src/Hal.c

```diff
@@ -26,11 +26,13 @@
 #include <Tic.h>
 #include <ConcurrentAccessFlag.h>
 
-#ifndef RT_MAX_HAL_SEMAPHORES
+#ifndef RT_MAX_HAL_SEMAPHORE
 #define RT_MAX_HAL_SEMAPHORES 8
 #endif
 
-#define NANOSECOND_IN_SECOND 1000000000u
+#define NANOSECOND_IN_SECOND 1000000000.0
+#define MAIN_CRYSTAL_OSCILLATOR_FREQUNECY 12
```

Comment:
If you have to define these values, please allow them to be redefined

Reply:
Done